runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -712,7 +776,36 @@ TR_J9ServerVM::isCloneable(TR_OpaqueClassBlock *clazzPointer)
 bool
 TR_J9ServerVM::canAllocateInlineClass(TR_OpaqueClassBlock *clazz)
    {
+   bool classInitialized;
+      {
+      OMR::CriticalSection getRemoteROMClass(_compInfoPT->getClientData()->getROMMapMonitor());
+      auto it = _compInfoPT->getClientData()->getROMClassMap().find((J9Class*) clazz);
+      if (it != _compInfoPT->getClientData()->getROMClassMap().end())
+         {
+         if (it->second.classInitialized)
+            {
+            return ((it->second.romClass->modifiers & (J9_JAVA_ABSTRACT | J9_JAVA_INTERFACE ))?
+                     true:false);
+            }
+         }
+      }
+
    JITaaS::J9ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   stream->write(JITaaS::J9ServerMessageType::VM_isClassInitialized, clazz);
+   classInitialized = std::get<0>(stream->read<bool>());
+
+   if (classInitialized)
+      {
+      OMR::CriticalSection getRemoteROMClass(_compInfoPT->getClientData()->getROMMapMonitor());
+      auto it = _compInfoPT->getClientData()->getROMClassMap().find((J9Class*) clazz);
+      if (it != _compInfoPT->getClientData()->getROMClassMap().end())
+         {
+            it->second.classInitialized = classInitialized;
+            return ((it->second.romClass->modifiers & (J9_JAVA_ABSTRACT | J9_JAVA_INTERFACE ))?
+                     true:false);
+         }
+      }
+
```

Comment:
If the class is not initialized even when you've just asked, then we can return false. No need to send another message.